src/NLog/Targets/Target.cs

```diff
@@ -722,6 +724,39 @@ protected string RenderLogEvent(Layout layout, LogEventInfo logEvent)
             }
         }
 
+        /// <summary>
+        /// Renders the event info in layout.
+        /// </summary>
+        /// <param name="layout">The layout.</param>
+        /// <param name="logEvent">The event info.</param>
+        /// <param name="defaultValue">The default value when <paramref name="layout"/> and/or <paramref name="logEvent"/> is <c>null</c></param>
+        /// <returns>Value representing log event.</returns>
+        protected T RenderLogEvent<T>([CanBeNull] Layout<T> layout, [CanBeNull] LogEventInfo logEvent, T defaultValue = default(T))
+        {
+            if (layout == null || logEvent == null)
+                return defaultValue;
+
+            if (OptimizeBufferReuse)
+            {
+                if (layout.IsFixed)
+                {
+                    return layout.RenderToValueOrDefault(logEvent, defaultValue);
+                }
+
+                if (TryGetCachedValue(layout, logEvent, out var value))
+                {
+                    return value;
+                }
+
+                using (var localTarget = ReusableLayoutBuilder.Allocate())
+                {
+                    return layout.RenderToValueInternal(logEvent, localTarget.Result, default(T));
+                }
+            }
+
+            return layout.RenderToValueOrDefault(logEvent, defaultValue);
```

Comment:
Should just call `layout.RenderToValue(logEvent, defaultValue);` (Have already checked layout is not-null)

Reply:
same as above, not possible (without cast).